tests/unit/setup-unit-test.ts

```diff
@@ -74,14 +76,42 @@ module.exports = async () => {
     path.join(vcastEnvPath, "unit.cpp")
   );
 
-  const clicastTemplateCommand = `cd ${vcastEnvPath} && ${clicastExecutablePath.trimEnd()} -l C template GNU_CPP11_X`;
-  const { stdout, stderr } = await promisifiedExec(clicastTemplateCommand);
-  if (stderr) {
-    console.log(stderr);
+  const clicastTemplateCommand = `cd ${vcastEnvPath} && ${clicastExecutablePath.trimEnd()} -l C template GNU_CPP11_X option VCAST_CODED_TESTS_SUPPORT TRUE`;
+
+  // Clicast execution
+  const { stdout: stdoutClicast, stderr: stderrClicast } =
+    await promisifiedExec(clicastTemplateCommand);
+  if (stderrClicast) {
+    console.log(stderrClicast);
     throw new Error(`Error when running ${clicastTemplateCommand}`);
   }
 
-  console.log(stdout);
+  console.log(stdoutClicast);
+
+  // Construct the path to the tool_version.txt file
+  const toolVersionPath = path.join(
+    clicastExecutablePath,
+    "..",
+    "DATA",
+    "tool_version.txt"
+  );
+  const toolVersion = fs.readFileSync(toolVersionPath).toString().trim();
+
+  // Activate coded tests support only for version 24
+  if (!toolVersion.startsWith("23")) {
+    const { stdout: stdoutClicastOption, stderr: stderrClicastOption } =
+      await promisifiedExec(
+        `cd ${vcastEnvPath} && ${clicastExecutablePath.trimEnd()} -lc option VCAST_CODED_TESTS_SUPPORT TRUE`
```

Comment:
Shall we make this a string, so we can log the string later on if there's an error?

Reply:
Refactored in cb0d4bcd97809b8bcf95bf9ab4e95a4b97fba3b6.